app/src/docs/components/PageTurner/index.jsx

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+import React from 'react'
+import { Link, withRouter, type Location } from 'react-router-dom'
+import cx from 'classnames'
+import { formatPath } from '$shared/utils/url'
+import SvgIcon from '$shared/components/SvgIcon'
+
+import type { NavigationLink } from '$docs/flowtype/navigation-types'
+
+import styles from './pageTurner.pcss'
+
+type Props = {
+    navigationItems: NavigationLink,
+    location: Location,
+}
+
+class PageTurner extends React.Component<Props> {
+    generateNavButtons() {
+        const { navigationItems } = this.props
+        const { pathname } = this.props.location
+
+        let pathMatch = false
+        let firstMatch = false
+
+        const navigationPaths = Object.entries(navigationItems).reduce((acc, [key, val], index) => {
```

Comment:
Needs comments about whats happening here 😄 Is this something that needs to happen on every render?

Reply:
Should be better now. I've split the render logic into the calculation (init class property) and a subrender function. 

Also made the naming less robotic and added some comments.

Reply:
Yep, much better 👍